staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -498,8 +503,76 @@ func GetFieldManagerFlag(cmd *cobra.Command) string {
 	return GetFlagString(cmd, "field-manager")
 }
 
+type DryRunStrategy int
+
+const (
+	DryRunNone DryRunStrategy = iota
+	DryRunClient
+	DryRunServer
+)
+
+func (d DryRunStrategy) Client() bool {
+	return d == DryRunClient
+}
+
+func (d DryRunStrategy) Server() bool {
+	return d == DryRunServer
+}
+
+func (d DryRunStrategy) None() bool {
+	return d == DryRunNone
+}
+
 func GetDryRunFlag(cmd *cobra.Command) bool {
```

Comment:
That name is not great since it's not clear that this is client side?

Reply:
I don't know if the function is actually needed, maybe you can call GetDryRunStrategy directly?

Reply:
Will do

Reply:
Thanks!